Revamp Service Bindings docs for RPC (#13252)
cloudflare/cloudflare-docs: content/workers/runtime-apis/rpc/reserved-methods.md

---
pcx_content_type: configuration
title: Reserved Methods
meta:
  title: Workers RPC — Reserved Methods

>   description: Reserved methods with special behavior that are treated differently

Reviewer: ```suggestion
  description: Reserved methods with special behavior that are treated differently.
```

---

# Reserved Methods

When extending `RpcTarget`, `WorkerEntrypoint`, and `DurableObject`, the following methods are reserved and cannot be redefined as RPC methods.

## `fetch()`

The `fetch()` method is treated specially — it can only be used to handle an HTTP request — equivalent to the [fetch handler](/workers/runtime-apis/handlers/fetch/).

> You may implement a `fetch()` method in your class that extends `WorkerEntrypoint` — but it must accept only one parameter of type [`Request`](https://developer.mozilla.org/en-US/docs/Web/API/Request), and must return an instance of [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response).

Reviewer: ```suggestion
You may implement a `fetch()` method in your class that extends `WorkerEntrypoint` — but it must accept only one parameter of type [`Request`](https://developer.mozilla.org/en-US/docs/Web/API/Request), and must return an instance of [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response), or a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) of one.
```

The caller may pass one or two parameters to `fetch()`. If the caller does not simply pass a single Request object, then a new Request is implicitly constructed, passing the parameters to its constructor, and that request is what is actually sent to the server.

> Some properties of Request control the behavior of fetch() on the client side and are not actually sent to the server. For example, the property `redirect: "auto"` (which is the default) instructs `fetch()` that if the server returns a redirect response, it should automatically be followed, resulting in an HTTP request to the public internet. The point is, fetch() makes an HTTP request according to the Fetch API standard.

Reviewer: ```suggestion
Some properties of `Request` control the behavior of `fetch()` on the client side and are not actually sent to the server. For example, the property `redirect: "auto"` (which is the default) instructs `fetch()` that if the server returns a redirect response, it should automatically be followed, potentially resulting in an HTTP request to the public internet. This behavior (and others like it) is preserved in Workers RPC — `fetch()` makes an HTTP request according to [the Fetch API standard](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).
```

---
pcx_content_type: configuration
title: Reserved Methods
meta:
  title: Workers RPC — Reserved Methods
  description: Reserved methods with special behavior that are treated differently
---

> # Reserved Methods

Reviewer 2: Could this whole page be merged into "visibility"? Feels related.
